File: statsig-go/feature_gate.go
package statsig

/*
#cgo CFLAGS: -I../statsig-ffi/include
#include "statsig_ffi.h"
#include <stdlib.h>
*/

type FeatureGate struct {
	Name              string            `json:"name"`
	Value             bool              `json:"value"`
	RuleID            string            `json:"rule_id"`
	EvaluationDetails EvaluationDetails `json:"details"`
	IdType            string            `json:"id_type"`
}

File: statsig-elixir/lib/statsig/feature_gate.ex
defmodule Statsig.FeatureGate do
  defstruct [
    :name,
    :value,
    :rule_id,
    :id_type
  ]

  @type t :: %__MODULE__{
          name: String.t(),
          value: boolean(),
          rule_id: String.t(),
          id_type: String.t()
        }
end
